ec/src/models/short_weierstrass_jacobian.rs

```diff
@@ -133,9 +113,11 @@ impl<P: Parameters> GroupAffine<P> {
             y2 == x3b
         }
     }
+}
 
-    /// Checks if `self` is in the subgroup having order that equaling that of
-    /// `P::ScalarField`.
+impl<P: Parameters> GroupAffine<P> {
+    /// Checks if `self` is in the subgroup having order equaling that of
+    /// `P::ScalarField` given it is on the curve.
     pub fn is_in_correct_subgroup_assuming_on_curve(&self) -> bool {
```

Comment:
Why aren't `is_on_curve()` and `is_in_correct_subgroup_assuming_on_curve()` part of the `AffineCurve` trait?

Reply:
Because they're supposed to represent points that are already on the curve and in the appropriate subgroup.